disk-buffering/src/main/java/io/opentelemetry/contrib/disk/buffering/internal/storage/utils/TimeProvider.java

```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.contrib.disk.buffering.internal.storage.utils;
+
+public class TimeProvider {
+  @SuppressWarnings({"FieldCanBeFinal", "FieldMayBeFinal"}) // Needed for tests
+  private static TimeProvider instance = new TimeProvider();
```

Comment:
can we pass the timeprovider in tests that need this instead of the static field?

Reply:
I think we can yeah, I remember I had to change this several times so maybe in the end it's not needed to have it as a static field. I'll double check.

Reply:
Thank you, I've made the changes.